[mountmuzzle] support custom BGM per mount

posimagi opened the pull request:
Allow users to set MountMuzzle BGM per mount per https://www.ffxiah.com/forum/topic/37379/lua-addon-ideas/29/#3700676.

New commands:
- `//mm set custom` (alias: `//mm s u`, for "user", to avoid conflict with "chocobo")
- `//mm c[ustom] MountName TrackNumber`

Features:
- Shows a success message when the user sets a track with the mount's name and track's title.
- Throws an error if the user tries to set a track for a mount that doesn't exist.

![image](https://github.com/Windower/Lua/assets/1747598/5ae509e6-2147-4d5d-a770-d411b7785f84)

Commit:
output_text_table renamed, made list, and output text updated.

z16 committed:
Merge pull request #2197 from vessellaj/dev

Yush: Added a VerboseKeys setting to enable key display in each macro set.

RubenatorX committed:
[tables] add table.pack

RubenatorX committed:
[Gearswap][Motes-Mappings]  Ltng. Threnody

There is no spell called "Lightning Threnody"

KenshiDRK committed:
[fields lib] fix wrong 0x0E8 packet direction

0x0E8 is an outgoing packet.

Byrth committed:
Merge pull request #2201 from KenshiDRK/field-lib

[fields lib] fix wrong 0x0E8 packet direction

z16 committed:
Packets/Fields: Updated outgoing 0x061 length

KenshiDRK committed:
[Slips lib] Update for June 2022

Added items to slips libs for the 10 June, 2022 FFXI Update.

z16 committed:
Merge pull request #2205 from KenshiDRK/Slips

[Slips lib] Update for June 2022

Roland-J committed:
Adds the texts:draggable() function

Allows users to toggle the draggable flag post-creation. Previously, this flag could only be passed in with the config in a images.new(config).

z16 committed:
Merge pull request #2207 from Roland-J/texts.draggable

Adds the texts:draggable() function

eLiidyr committed:
Update fields.lua

Added Current TVR Mission.

eLiidyr committed:
Update fields.lua

Correct TVR `_junk` field.

z16 committed:
Merge pull request #2208 from eLiidyr/dev

Update fields.lua

KenshiDRK committed:
[Slips lib] Update for July 2022

Added items to slips libs for the 10 July, 2022 FFXI Update.

z16 committed:
Merge pull request #2211 from KenshiDRK/Slips

[Slips lib] Update for July 2022

bsb002-flash-tester committed:
Allow multiple 'text' children in a single XML tag.

RubenatorX committed:
[dialog.lua] Fix occasional missing \ at the end

Byrth committed:
Merge pull request #2217 from Windower/dev

Slips update

KenshiDRK committed:
[slips lib] August 10, 2022 FFXI Update

Adding items to slips from the august 10 update

z16 committed:
Merge pull request #2219 from KenshiDRK/Slips

[slips lib] August 10, 2022 FFXI Update

funkworkz committed:
Update setbgm.lua

Song name released via ffxi official YouTube channel video

z16 committed:
Merge pull request #2220 from funkworkz/patch-5

Update setbgm.lua

xurion committed:
Update EmpyPopTracker to 2.7.0

z16 committed:
Merge pull request #2221 from xurion/dev

Update EmpyPopTracker to 2.7.0

trv6 committed:
Remove strings method

Strings isn't required

z16 committed:
Merge pull request #2222 from trv6/patch-2

Remove strings method from libs/dialog

Imisnew2 committed:
Update fields.lua: Update 0x0E0 (Linkshell Equip)

Added definition for which bag the linkshell is being equipped from.

z16 committed:
Merge pull request #2223 from Imisnew2/patch-1

Update fields.lua: Update 0x0E0 (Linkshell Equip)

Nifim committed:
Set label back to 'Inventory Slot'

Slot is a better name but we dont want to break existing code that relies on the name. This could be revisited later

lili-ffxi committed:
Added new currencies.

The August 2022 update to the game added 4 new fields to the currencies menu. This update reflects that.

Nifim committed:
Merge pull request #2224 from lili-ffxi/patch-3

Added new currencies.

Chiaia committed:
Merge pull request #2226 from Windower/dev

Dev -> Live

Chiaia committed:
Preemptive for Sept 2022 Update

Will need double checking that SE didn't screw up and for the Slip Item ID to be added.

Chiaia committed:
Amini Bragues +2 23296 Added

Byrth committed:
digger messages updates

Byrth committed:
boxdestroyer message updates

Byrth committed:
Merge pull request #2227 from Windower/sept-2022-messages-update

Sept 2022 messages update

Chiaia committed:
Merge pull request #2228 from Windower/dev

Dev -> Live

Chiaia committed:
Sept 11th, 2022 Update

Chiaia committed:
Merge pull request #2229 from Windower/dev

Sept 11th, 2022 Update Slips.lua

Chiaia committed:
Fixes + Slip 22 added items.

Chiaia committed:
Merge pull request #2230 from Windower/dev

Fixes + Slip 22 added items.

Chiaia committed:
Slip 29 Support

Chiaia committed:
Merge pull request #2231 from Windower/dev

Slip 29 Support

Byrth committed:
Add Blueblade Fell

Byrth committed:
Merge pull request #2235 from Windower/update-slips-2022-10

Add Blueblade Fell

Byrth committed:
Merge pull request #2236 from Windower/dev

dev -> live for slips

funkworkz committed:
Update setbgm.lua

Unknown Title. If someone has a better idea for a placeholder name, by all means please edit it.

WinterSolstice8 committed:
[shortcuts] Pass valid commands through if target and command is valid

zetaepyon committed:
Create linuxfix.lua

Chiaia committed:
Merge pull request #2237 from funkworkz/patch-6

Update setbgm.lua

Chiaia committed:
Slip 30

Chiaia committed:
Merge pull request #2240 from Windower/dev

Dev -> Live

Byrth committed:
Merge pull request #2238 from WinterSolstice8/fix_exact_commands_with_low_player_IDs

[shortcuts] Pass valid commands through if target and command is valid

funkworkz committed:
Update setbgm.lua

Learned the song name from the new Orchestrion Key Item implemented in game.

z16 committed:
Merge pull request #2242 from funkworkz/patch-7

Update setbgm.lua

Nifim committed:
Merge pull request #2216 from Imisnew2/imisnew-config-multiple-text-children

Config: Allow multiple 'text' children in a single XML tag.

Chiaia committed:
Slip 30 Support

Chiaia committed:
Merge pull request #2248 from Windower/dev

Dev -> Live

bsb002-flash-tester committed:
Revert "[shortcuts] Pass valid commands through if target and command is valid"

This reverts commit 3c94fd5613fabad7215173d0d497a310df48e5e9.

RubenatorX committed:
Merge pull request #2249 from Imisnew2/revert-shortcuts-change

Revert "[shortcuts] Pass valid commands through if target and command is valid."

RubenatorX committed:
Merge pull request #2250 from Windower/dev

revert breaking [shortcuts] change

RubenatorX committed:
revert breaking [config] changes

RubenatorX committed:
revert breaking [config] change -- but for real this time

RubenatorX committed:
Merge pull request #2251 from Windower/dev

revert breaking [config] changes

Chiaia committed:
Missing space fixed

Chiaia committed:
Merge pull request #2255 from Windower/dev

Dev - >Live

Byrth committed:
Boxdestroyer update

Byrth committed:
Merge pull request #2256 from Windower/update-boxdestroyer-2022-12

Boxdestroyer update 2022-12

Chiaia committed:
Merge pull request #2257 from Windower/dev

Dev -> Live

Byrth committed:
update slips for 2022-12

Byrth committed:
Merge pull request #2258 from Windower/2022-12-slips

update slips for 2022-12

Byrth committed:
Merge pull request #2259 from Windower/dev

Slips update!

posimagi committed:
[craft] report quantity of items crafted

posimagi committed:
[craft] report quantity of items crafted

posimagi committed:
[craft] Move multi-recipes (those that use Bundling Twine, Spindle, etc.) to higher recipe numbers than the standard recipe that makes the basic quantity. Fix issues with several recipes that erroneously show multiple copies of the bundling guild item.

posimagi committed:
[craft] Move multi-recipes (those that use Bundling Twine, Spindle, etc.) to higher recipe numbers than the standard recipe that makes the basic quantity. Fix issues with several recipes that erroneously show multiple copies of the bundling guild item.

Byrth committed:
Add queue library

Byrth committed:
Add queue library

Byrth committed:
Batch buff_details and add for party members

Byrth committed:
Batch buff_details and add for party members

Byrth committed:
Fix bug in move_spell_target()

Byrth committed:
Fix bug in move_spell_target()

Byrth committed:
add disable check to check_spell() for gear spells

Byrth committed:
add disable check to check_spell() for gear spells

Byrth committed:
I just deleted dead code

Byrth committed:
I just deleted dead code

Byrth committed:
un-disable buff_refresh events

Byrth committed:
un-disable buff_refresh events

Byrth committed:
add comment

Byrth committed:
add comment

Byrth committed:
Merge pull request #2265 from Windower/gearswap-cpu-no-longer-overheats

Gearswap cpu no longer overheats

Byrth committed:
Make send no longer error when logged out

Byrth committed:
Make send no longer error when logged out

Byrth committed:
Merge pull request #2266 from Windower/fix-send-bug

Make send no longer error when logged out

posimagi committed:
[craft] Upgraded create_recipes.py to python3 and made it robust enough to not break on recipe(s) with unknown ingredients (though the output is not useful--not that it could be crafted anyway).

posimagi committed:
[craft] Upgraded create_recipes.py to python3 and made it robust enough to not break on recipe(s) with unknown ingredients (though the output is not useful--not that it could be crafted anyway).

posimagi committed:
[craft] generate new recipes.lua

posimagi committed:
[craft] generate new recipes.lua

Byrth committed:
Merge pull request #2267 from posimagi/craft-python3

[craft] repair and upgrade create_recipes.py to python3

KenshiDRK committed:
[Slips lib] Update for new items

Adding the new items Cornelia's belt, Kni cap and Knit cap +1

KenshiDRK committed:
[Slips lib] Update for new items

Adding the new items Cornelia's belt, Kni cap and Knit cap +1

KenshiDRK committed:
[Battlemod] Antidedupliction fix

Antideduplication fix when buffs with the same name wear off at the same time. Some more minor fixes.

KenshiDRK committed:
Fix for unsimplified Resist! messages

Fix for unsimplified Resist! messages

bsb002-flash-tester committed:
Update 0x03C to have starting index field.

bsb002-flash-tester committed:
Update 0x03C to have starting index field.

Byrth committed:
Update slips for the 2023-02 update

Byrth committed:
Update slips for the 2023-02 update

lili-ffxi committed:
Fix Lightningday spelling

Nifim committed:
Merge pull request #2276 from lili-ffxi/patch-4

Fix Lightningday spelling

lili-ffxi committed:
Update fields.lua

The new field in 0x0119 contains the difference between the base recast of a job ability, and the actual recast as affected by merits/gear/job points.

z16 committed:
Merge pull request #2279 from lili-ffxi/patch-5

Update fields.lua

Byrth committed:
an unfathomably wondrous contribution

KenshiDRK committed:
[Slips lib] Adding new item

Adding Platinum Moogle Belt to the slips lib.

Byrth committed:
Update user_functions.lua

LordTreyBennett committed:
Update README.md

corrected the English spelling and grammar from "if you won't" to the correct "if you do not want" so that the meaning of the sentence matches the command mentioned and corrected the spelling of want from won't, meaning will not instead of want's meaning of an expectation of action.

Byrth committed:
Remove XML option for //gs export

funkworkz committed:
Update setbgm.lua

Anticipated new songs for May 2023 update. Placeholders are already in the DATs.

Nifim committed:
Merge pull request #2294 from funkworkz/patch-8

[setbgm] Anticipated new songs for May 2023 update

KenshiDRK committed:
[Slips lib] Adding new items

Added Ageist and Regis to storage slip 22

z16 committed:
Merge pull request #2283 from KenshiDRK/Slips

[Slips lib] Adding new item

DrakeFS committed:
Update Mote-Utility.lua

Changed book error text, in set_macro_page(), to reflect that there are 40 macro books.

z16 committed:
Merge pull request #2295 from DrakeFS/dev

[GearSwap] Update error text in Mote-Utility.lua

z16 committed:
Merge pull request #2292 from Windower/gs-export-remove-xml

Remove XML option for //gs export

z16 committed:
GearSwap: Added minify feature to export options

KenshiDRK committed:
Fixing some statuses

Fixing some statuses to use the long form.

z16 committed:
Merge pull request #2272 from KenshiDRK/battlemod

[Battlemod] Antideduplication fix

RubenatorX committed:
Update slips.lua Poison Axe (+1) -- guess

RubenatorX committed:
[packets] fields.lua Zone Line Packet discoveries

maxom1310 committed:
Update helper_functions.lua

maxom1310 committed:
Update helper_functions.lua

RubenatorX committed:
Merge pull request #2296 from maxom1310/patch-1

[Gearswap] Properly handle new song "Aria of Passion"

terrawheat committed:
Update gametime.lua to resolve issue with loading on Lightningday

The gametime addon will not load correctly on Lightningday due to a check only looking for the pluralised form.

z16 committed:
Merge pull request #2298 from terrawheat/patch-1

Update gametime.lua to resolve issue with loading on Lightningday

Byrth committed:
make it possible to access text settings again

z16 committed:
Merge pull request #2299 from Windower/fix-pointwatch

make it possible to access text settings again

KenshiDRK committed:
[Slips] Update for July 10, 2023 items

Adding items from the July 10, 2023 update.

z16 committed:
Merge pull request #2300 from KenshiDRK/Slips

[Slips] Update for July 10, 2023 items

lili-ffxi committed:
Update itemizer.lua

This change fixes an issue where itemizer would fail to locate HQ items, whose name ended with "+1", "+2" etc, because was treating the the +x at the end of the string as the amount argument instead.

lili-ffxi committed:
Update setbgm.lua

Added `find` argument and corresponding functionality to search song names for a provided string.

Nifim committed:
Merge branch 'live' into dev

# Conflicts:
#	addons/GearSwap/helper_functions.lua
#	addons/GearSwap/user_functions.lua
#	addons/libs/slips.lua

Nifim committed:
Remove unnecessary print

Nifim committed:
Merge pull request #2303 from Windower/dev

Merge Dev to Live

Byrth committed:
2023-08 boxdestroyer update

Byrth committed:
2023-08 boxdestroyer update

AkadenTK committed:
Fix for allowing blocked messages through

z16 committed:
Merge pull request #2308 from AkadenTK/dev

[Addon][translate] Fix for allowing blocked messages through

lili-ffxi committed:
Update setbgm.lua

The `find` argument will print a notice if no matching songs are found.

z16 committed:
Merge pull request #2302 from Windower/lili-ffxi-patch-1

Update setbgm.lua

KenshiDRK committed:
[Slips lib] August 2023 Update

August 2023 Update for the Slips library.

funkworkz committed:
Update setbgm.lua

Nifim committed:
Merge pull request #2310 from funkworkz/dev-2

Update setbgm.lua

lili-ffxi committed:
Update setbgm.lua

A value of `0` effectively disables the music reproduction for the given type. Debated calling it "silence" or "disabled" but settled for "No Music" in the end.

Also bumped version after this change and last PR by Funk.

KenshiDRK committed:
[Slips lib] Forget to add slip 31 id

Adding Slip 31 id

z16 committed:
Merge pull request #2311 from Windower/lili-ffxi-patch-1

Update setbgm.lua

lili-ffxi committed:
Added functionality to Itemizer (#2312)

* Added functionality to Itemizer

Significant changes to Itemizer:
- added auto-stacking of the destination bag when moving partial stacks of stackable items, along with a setting for the functionality, and a manual argument to invoke it.
- changed the previous fix to the recognition of high quality item names (ending in `+1`, `+2`, etc) to include any other non-digit character (i.e `#`)
- changed help text to reflect the new functionality (and fixed capitalization)
- changed all instances of `windower.add_to_chat()` to use `log()` instead, since it's using the `luau` library anyway
- changed version-recognizing logic so that it recognizes changes in major and minor versions automatically without the need for hardcoded values
- Adjusted version check so that it only fires when the user is logged in all the way and thus the chatlog is available to print the notice.
- bumped version number

I am, tentatively, not using the packets library for the stacking functionality, since it adds a good bit of overhead and there can be situations where the addon might be spamming stacking commands, especially when the //put commands are invoked via scripts - as it's common. On the other hand, the addon is already using item resources which is a lot of overhead on its own, so I am not sure if it would be better to use the packets lib as well at that point. I'm open to suggestions.

lili-ffxi committed:
Bugfix after commit #2312

`log()` can translate booleans directly into strings whereas `string.format()` doesn't.

Byrth committed:
add user_pcall variant that doesn't throw

KenshiDRK committed:
[Slips]  September 11th, 2023 FFXI Update

Added items from the September 11th. 2023 FFXI Update.

eLiidyr committed:
Addon: Shortcuts

If a user attempts to use the default in-game item command, shortcuts attempts to cast a spell if the spell and item name are the same.

Block shortcuts from finishing command logic if it detects default item command method.

eLiidyr committed:
Addon: Shortcuts.

Adjusted the pattern, and forgot to add a space pattern check for item names with spaces.

z16 committed:
Merge pull request #2317 from eLiidyr/patch-2

Addon: Shortcuts

eLiidyr committed:
Addon: Shortcuts

Revert earlier commits.

eLiidyr committed:
Addons: Shortcuts [ statics.lua ]

Add /item to unhandled command list.

RubenatorX committed:
Merge pull request #2318 from eLiidyr/dev

Addons: Shortcuts

RubenatorX committed:
Merge pull request #2316 from KenshiDRK/Slips

[Slips]  September 11th, 2023 FFXI Update

nmarigoni committed:
[barfiller] Fix EXP rate calculation (#2319)

* Update statics.lua

Updated to use current time (os.time() ) rather than seconds of CPU time (os.clock() ) to calculate elapsed time.

Added msg 253 to parsing to capture EXP awarded from EXP chains.

* Update barfiller.lua

Update to use current time rather than CPU seconds

KenshiDRK committed:
[Slips] October 10th, 2023 FFXI Update

Added items from the September 11th. 2023 FFXI Update.

z16 committed:
Merge pull request #2320 from KenshiDRK/Slips

[Slips] October 10th, 2023 FFXI Update

RubenatorX committed:
Update chat.icons

Add Mentor Flags and Information icon

RubenatorX committed:
[boxdestroyer] Update messages.lua

RubenatorX committed:
Merge branch 'live' into dev

RubenatorX committed:
[JobChange] allow for uppercase in commands

RubenatorX committed:
[boxdestroyer] Update messages.lua

RubenatorX committed:
[JobChange] remove unnecessary repeated lower() calls

RubenatorX committed:
[JobChange] remove unnecessary repeated lower() calls

KenshiDRK committed:
[Slips] Swapping wrong item order

Swapping order for hunter's braccae and socks.

Nifim committed:
Merge pull request #2325 from KenshiDRK/Slips

[Slips] Swapping wrong item order

RubenatorX committed:
[SpellCheck] Update SpellExceptions.lua

the id for Tonko: San -- an unlearnable spell -- was replaced by Siren

lili-ffxi committed:
Update author field for battlemod.lua

z16 committed:
Merge pull request #2327 from Windower/lili-ffxi-patch-1

Update author field for battlemod.lua

lili-ffxi committed:
Update extdata.lua

Add 'Rank' field to augment system 4 (Ultimate Weapons, Odyssey Armor, Dynamis Divergence Weapons and Necks, Unity Rewards)

z16 committed:
Merge pull request #2328 from Windower/lili-ffxi-patch-1

Update extdata.lua

lili-ffxi committed:
Update extdata.lua

Partial revert to avoid a potentially breaking change.

z16 committed:
Fixed various issues with resources

z16 committed:
Merge pull request #2330 from Windower/feature/resources-lib-fixes

Fixed various issues with resources

z16 committed:
Merge pull request #2329 from Windower/lili-ffxi-patch-1

Update extdata.lua

z16 committed:
Libs/Resources: Fixed flag parsing

z16 committed:
Merge pull request #2331 from Windower/feature/resources-lib-fixes

Libs/Resources: Fixed flag parsing

z16 committed:
Libs/Resources: Amended behavior when filtering resources by an attribute

z16 committed:
Merge pull request #2332 from Windower/feature/resources-lib-attribute-filter

Libs/Resources: Amended behavior when filtering resources by an attribute

z16 committed:
Lib/Strings: Added encoding aware iteration (#2333)

* Lib/Strings: Added encoding aware iteration

* Libs/Strings: Refactored string.it

* Lib/Strings: Fixes/cleanups

z16 committed:
Lib/Tabls: Added iterator packing function (#2334)

lili-ffxi committed:
Update strings.lua (#2335)

Bugfix in the new iterator

z16 committed:
Libs/Strings: Added encoding aware string.find, string.match, string.gmatch and string.sub (#2336)

z16 committed:
Lib/Strings: Fixed argument parser for new string functions

z16 committed:
Lib/Strings: Fixed argument processing

KenshiDRK committed:
[slips] Updating new items (#2337)

Updating new items for the slips lib

RubenatorX committed:
[Clock] display settings bugfix

KenshiDRK committed:
[Slips] Adding new items (#2343)

Adding Esoteric Athame and Eyre Cap to Storage slips 22

lili-ffxi committed:
Update send.lua (#2339)

* Update send.lua

Adjust string processing to reinstate double quotes removed by the command processor.

* Update send.lua

* Update send.lua

Spaces after commas.

* Update send.lua

Fixed an oversight around the `<tid>` functionality.

* Update send.lua

Added toggle command for the debug functionality.

* Update send.lua

* Update send.lua

Rename dbg to debug

lili-ffxi committed:
Update send.lua (#2344)

* Update send.lua

Extensive code revision
Added two new targets: @party and @zone

* Update send.lua

All commas are now properly married to a space each

* Update send.lua

Removed all trailing spaces

z16 committed:
Libs/Strings: Added encoding-aware string.split

z16 committed:
Libs: Various string related fixes

lili-ffxi committed:
Update send.lua (#2345)

* Update send.lua

Initial refactoring by Arcon

* Update send.lua

Tentative fix for a couple quirks caused by splitting the `target` arg off `...`

* Update send.lua

Added obligatory `<on|off>` argument to `@debug`
Moved debug prints around for simplicity
Fixed bugs:
- a stray `:upper()` had survived the refactoring
- a `@job` target that was different from the sender main job would be lost in the void and not passed on

* Update send.lua

z16 committed:
Updated .gitignore

z16 committed:
Libs/Tables: Adjusted table functions to iterate generically

zmsl committed:
Fixed 0x00A field typo

RubenatorX committed:
Merge pull request #2348 from zmsl/lib/fields-fix

(Libs/Packets) Fixed 0x00A field typo

posimagi committed:
[pointwatch] support true master levels

posimagi committed:
[pointwatch] clean up tnml table

posimagi committed:
[pointwatch] update version number

posimagi committed:
[pointwatch] create a new var ep.true_master_level to support display of true master level under level sync without removing synced master level option

posimagi committed:
[pointwatch] add "ep" format string for watching only EP

posimagi commented:
Bump.

posimagi committed:
[pointwatch] make true master level the default

RubenatorX committed:
Merge pull request #2350 from posimagi/pointwatch-real-ml

[pointwatch] support the display of true ML under level sync

lili-ffxi committed:
Update helper_functions.lua (#2352)

Preserve numerical values for spell.skill and spell.element as they're potentially useful in multiple scenarios, most often when comparing stuff against resources

posimagi committed:
[pointwatch] minor fix for EP table value (#2353)

RubenatorX committed:
[boxdestroyer] Update messages.lua

z16 commented on addons/MountMuzzle/mountmuzzle.lua:
```diff
@@ -258,4 +286,14 @@ windower.register_event('incoming chunk', function(id, data)
     end
+end)
+
+windower.register_event('outgoing chunk',function(id, data)
```
Space after the comma.

z16 commented on addons/MountMuzzle/constants.lua:
```diff
@@ -99,13 +100,47 @@ muzzles = {
     }
 }
 
+songs = {
```
Is this the same list that `setbgm` uses? If so, it would be nice to combine them somewhere. Honestly, this looks like something for resources (probably added in the `fixes.xml`). But shouldn't stop this PR, just something to do in the future, so that we don't have to maintain both lists.

RubenatorX replied:
@z16 
There may be an argument for them to be one addon instead of two since, they both simply change game music. :shrug:

lili-ffxi replied:
I haven't read the PR enough to remember if it actually does it, but there were talks of making this addon also able to select a *random* BGM to go along with the random mount. Setbgm wouldn't be able to do that without some refactoring. Besides, associating a specific different song to each mount also feels out of scope for setbgm, but ymmv.

Anyway, +1 to adding custom background music resources. I can even take care of that!

posimagi replied:
This is indeed the same list as setbgm. I'm not at all opposed to combining them and/or moving the track list into resources, etc.

z16 commented on addons/MountMuzzle/mountmuzzle.lua:
```diff
@@ -87,11 +91,17 @@ local help = {
+local possible_mounts = L{}
+for _, mount in pairs(_resources.mounts) do
+    possible_mounts:append(mount.name:lower())
+end
```
Can condense these lines to this:
```lua
local possible_mounts = _resources.mounts:map(function(mount) return mount.name:lower() end)
```

z16 commented on addons/MountMuzzle/constants.lua:
```diff
@@ -62,6 +62,7 @@ music = {
         mount = 84,
         chocobo = 212,
         zone = 0,
+        custom = 9998
```
I assume these are song IDs and you uses `9998` just to make sure it doesn't exist, and because `9999` was already taken? If so, would it be an issue if they both uses `9999`? In general, I'd prefer clearly separate numbers, like negative numbers, or ideally just leaving them `nil`, if that works with the rest of the code.

posimagi replied:
I don't think there'd be any issue with making it also 9999. I will test it out.

z16 reviewed (changes requested):
Left some notes, the only thing that needs to change is the missing space, the rest is optional and I'll leave to you.

KenshiDRK committed:
[Slips lib] Adding Dispatcher's Axe (#2355)

Adding new item Dispatcher's Axe to storage slips 22

lili-ffxi committed:
Add Treat Staff II to item list. (#2356)

lili-ffxi committed:
Fix typo (#2357)

Forgot some quotes because of course I did.

KenshiDRK committed:
[Slips] Adding June 2024 FFXI update item

Adding Arrogance set to the Storage slips 31

KenshiDRK committed:
Merge pull request #2359 from KenshiDRK/Slips

[Slips] Adding June 2024 FFXI update item

lili-ffxi committed:
Fixes bug around /bstpet prefix. (#2358)

In-game, /bstpet accepts both a number (an index of the pet move to perform) or the full name of the desired move. Previously gearswap would only trigger when the index was provided and now with the full name. This change fixes triggers.lua to trigger gearswap when /bstpet is used with the pet ability name.

lili-ffxi committed:
Update strings.lua (#2361)

* Update strings.lua

Include ffxi's specific color codes and autotranslate delimiters

* Update strings.lua

More tweaks to characters, and restore trailing empty token.

lili-ffxi committed:
Update SpellBook.lua (#2362)

Fixes following the changes to strings library.

lili-ffxi committed:
Fix elemental resist entries in fields.lua (#2365)

Somehow somewhen these fields have been shuffled. Now they're fixed.

svanheulen committed:
Update texts.lua (#2366)

Fixes texts.hover when text is right justified

KenshiDRK committed:
[Slips] Adding July and August Updates

Adding Apathy and Rage sets to Storage Slips 31.

RubenatorX committed:
Merge pull request #2367 from KenshiDRK/Slips

[Slips] Adding July and August Updates

KenshiDRK committed:
[FindAll] Update to dynamically add storage slips

Update to dynamically add storage slips

RubenatorX committed:
Merge pull request #2368 from KenshiDRK/Findall

[FindAll] Update to dynamically add storage slips

RubenatorX committed:
Update dialog.lua

Fix for instances

lili-ffxi committed:
Fix potential integer overflow when displaying very large amounts of damage (#2370)

Scoreboard would overflow and display a negative number once the total amount of damage dealt was over 2^32. Changing `%d` for `%.0f` where appropriate to avoid this.

lili-ffxi committed:
[gearswap] Parse last_incoming packets in chronological order. (#2371)

* Parse last_incoming packets in chronological order.

Previously gearswap would present stale player data right upon loading, due to reading last_incoming packets out of chronological order. This commit fixes that.

* Version bump.

* Fix potential timestamp clash

Rewrote the function to prevent issues in case of two packets having the same timestamp.

lili-ffxi committed:
Update labels on Item flags 02 and 03 (#2369)

* Update labels on Item flags 02 and 03

As per https://github.com/Windower/POLUtils/issues/25

I am not sure about naming (tho these match the labels used on BG-Wiki, and I think LSB too), opinions appreciated.

* Rename Flag00 to Wall-mounted.

Using `Wall-mounted` because that's what FFXIV calls these.

eLiidyr committed:
New add-on: Autoex (Autoexec replacement). (#2071)

svanheulen committed:
Update extdata.lua (#2373)

Add augment values for "pet: PDT" and "pet: MDT" on Ambuscade capes

lili-ffxi committed:
[gearswap] Fix issue with packet initialization. (#2374)

* Update statics.lua

* Update gearswap.lua

KenshiDRK committed:
[Slips] Adding September 2024 FFXI update item

Adding Cowardice set to storage slips 31

KenshiDRK committed:
Merge pull request #2375 from KenshiDRK/Slips

[Slips] Adding September 2024 FFXI update items

cinderborn committed:
Fixed xiv bar to use Windower UI scale values. (#2253)

Co-authored-by: Serena <sylfschiffer@gmail.com>

KenshiDRK committed:
[Slips] Adding October 2024 FFXI update item (#2378)

Adding Envy set to storage slips 31

KenshiDRK committed:
[Battlemod] Fix for Fighting Stances moves (#2379)

This fix the Fighting Stances moves to show the greek letters.

KenshiDRK committed:
[Shortcuts] Fix for Monstrosity Fighting Stances (#2380)

* [Shortcuts] Fix for Monstrosity Fighting Stances

This allow the use of Fighting Stance commands in monstrosity through shortcuts.

* Update shortcuts.lua

posimagi commented:
Updated to address comments.

posimagi committed:
Merge remote-tracking branch 'upstream/dev' into dev

posimagi committed:
Merge remote-tracking branch 'upstream/dev' into dev

posimagi committed:
[mountmuzzle] Allow users to set MountMuzzle BGM per mount
